doc/api/process.markdown

```diff
@@ -7,14 +7,21 @@ It is an instance of [`EventEmitter`][].
 
 ## Event: 'beforeExit'
 
-This event is emitted when Node.js empties its event loop and has nothing else to
-schedule. Normally, Node.js exits when there is no work scheduled, but a listener
-for `'beforeExit'` can make asynchronous calls, and cause Node.js to continue.
+This event is emitted when Node.js empties its event loop and has nothing else
+to schedule. Normally, Node.js exits when there is no work scheduled, but a
+listener for `'beforeExit'` can make asynchronous calls, and cause Node.js to
+continue.
 
-`'beforeExit'` is not emitted for conditions causing explicit termination, such as
-[`process.exit()`][] or uncaught exceptions, and should not be used as an
+`'beforeExit'` is not emitted for conditions causing explicit termination, such
+as [`process.exit()`][] or uncaught exceptions, and should not be used as an
 alternative to the `'exit'` event unless the intention is to schedule more work.
 
+## Event: 'disconnect'
+
+If Node.js was spawned with an IPC channel, this event will be emitted when
+either [`process.disconnect()`][] or [`ChildProcess.disconnect()`] in parent
+process is called.
+
```

Comment:
It was not documented. Though `child_process` is responsible for patching `process` object when one Node process spawns another Node process... In `process` docs we have currently several such "additions" described (`send()`, `message` event), so I thought it might be good to add `disconnect` event description too.

And if you're talking about code:
1. [Master adds `NODE_CHANNEL_FD` env var](https://github.com/nodejs/node/blob/a06066ca3180c86fa93b4f220867c1fd0c54f678/lib/internal/child_process.js#L268)
2. [Child checks for this env var](https://github.com/nodejs/node/blob/a06066ca3180c86fa93b4f220867c1fd0c54f678/src/node.js#L826) and [sets up messaging, if found](https://github.com/nodejs/node/blob/a06066ca3180c86fa93b4f220867c1fd0c54f678/src/node.js#L840).
3. Finally [setupChannel() is called by child_process._forkChild()](https://github.com/nodejs/node/blob/a06066ca3180c86fa93b4f220867c1fd0c54f678/lib/child_process.js#L63) and [there](https://github.com/nodejs/node/blob/a06066ca3180c86fa93b4f220867c1fd0c54f678/lib/internal/child_process.js#L663) things are set up that when child is disconnected it emits `'disconnect'` event.